pkg/dock/dock.go

```diff
@@ -134,9 +136,13 @@ func (ds *dockServer) CreateVolume(ctx context.Context, opt *pb.CreateVolumeOpts
 
 // DeleteVolume implements pb.DockServer.DeleteVolume
 func (ds *dockServer) DeleteVolume(ctx context.Context, opt *pb.DeleteVolumeOpts) (*pb.GenericResponse, error) {
+	ds.mutex.Lock()
 	// Get the storage drivers and do some initializations.
 	ds.Driver = drivers.Init(opt.GetDriverName())
-	defer drivers.Clean(ds.Driver)
+	defer func() {
+		drivers.Clean(ds.Driver)
+		ds.mutex.Unlock()
```

Comment:
only driver clean needs to be protected?
how about ds.Driver.DeleteVolume()? 
Also, please check the lock and unlock are consistent. we are locking and unlocking inside the defer function. safe? kindly check

Reply:
Can there be a case that Init/Clean make the function return form there only? Basically deferred function not getting called/called properly, In that case what about the locked mutex